EL-3150 - Tab Set

ashley-hunter opened the pull request:
- Created a new WCAG compliant tabset component
- Added documentation example
- Added protractor tests

donna-marijne commented on src/components/tabset/tabset.component.html:
```diff
@@ -0,0 +1,45 @@
+            (focus)="tabset.focused$.next(true)"
+            (blur)="tabset.focused$.next(false)"
+            (mousedown)="tabset.select(tab)"
+            (keydown.arrowleft)="tabset.selectAtIndex(index - 1); $event.preventDefault()"
```
Possibly need some logic here to skip over disabled tabs?

donna-marijne commented on src/components/tabset/tabset.component.ts:
```diff
@@ -0,0 +1,32 @@
+})
+export class TabsetComponent {
+
+    @Input() type: 'tabs' | 'pills' = 'tabs';
```
Do we support this? Should probably document or remove.

donna-marijne commented on src/components/tabset/tabset.component.ts:
```diff
@@ -0,0 +1,32 @@
+
+    @Input() type: 'tabs' | 'pills' = 'tabs';
+    @Input() justified: boolean = false;
+    @Input() vertical: boolean = false;
```
Also document or remove.

donna-marijne commented on src/components/tabset/tabset.component.html:
```diff
@@ -0,0 +1,45 @@
+    [class.minimal-tab]="minimal"
+    [attr.aria-label]="ariaLabel">
+
+	<li role="presentation" 
```
I think if you set this to `role="none"` then it will remove the "list item" role implied by the li element. Current this is dictated as "1 of 4 list item."

ashley-hunter commented:
1. Fixed
2. Non active tab content now uses aria-hidden - ngIf loses tab state when they are changed so I didn't use that
3. This should now be fixed, I believe the issue was the focusIf directive which kept refocusing the active item. Now replaced with an internal tabFocus directive that handles focus more appropriately
4. Removed justified as I realised our styles did not fully support it
5. Added deprecatedFor
6. Removed customize region from snippet
7. Tab selection now takes into account disabled tabs
8. Removed tab type - our styles don't support it
9. Removed vertical - our styles don't support it
10. Tried role="none", but that didn't seem to fix the issue. From what I can tell the only way would be to change it to not be ul & li - however this breaks all bootstrap styling and our styling as it all explicitly uses selectors like `.nav > li > a`. I can redo the tab styles if you prefer as but they will no longer be 'bootstrap' tabs.

ashley-hunter commented:
Fixed customize section
